src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,223 +37,281 @@
+                "The country configuration file does not contain"
+                "any region name. Please update and try again."
+            )
 
-    Returns
-    -------
-    country_data : str
-        if data found return the output_dir, else return None
-    """
-    url_country_date = _BASE_URL_COUNTRY.format(
-        iso2=iso2.upper(), YYYY=date_pub.year, MM=date_pub.month
-    )
-
-    # filenames have upper iso2, so use that for dirs as well
-    output_dir_country = (
-        output_dir / f"{iso2.upper()}{date_pub.strftime('%Y%m')}"
-    )
-    zip_filename = f"{output_dir_country.name}.zip"
-    if not output_dir_country.exists() or use_cache is False:
-        country_data = _download_zip(
-            url=url_country_date,
-            zip_filename=zip_filename,
-            output_dir=output_dir_country,
+        self._region_name = self._country_config.fewsnet.region_name
+        self._region_code = _REGION_NAME_CODE_MAPPING[self._region_name]
+
+    # mypy will give error Signature of "download" incompatible with supertype
+    # "DataSource" due to `date_pub` not being an arg in `DataSource`. This is
+    # however valid so ignore mypy
+    def download(  # type: ignore
+        self,
+        date_pub: str,
+        clobber: bool = False,
+    ) -> Optional[Path]:
+        """
+        Retrieve the raw FEWS NET data.
+
+        Depending on the region and date, this data is published per region or
+        per country. This function retrieves the country data
+        if it exists, and else the regional data for `date_pub`.
+
+        Parameters
+        ----------
+        date_pub : str
+            date for which the data should be downloaded
+            only the year and month part are used
+            this commonly refers to the month of the Current Situation period
+        clobber : bool, default = False
+            If True, overwrites existing raw files
+
+        Examples
+        --------
+        >>> download(date_pub="2020-10-01")
+        """
+        date_pub_datetime = parse(date_pub)
+
+        # we prefer the country data as this more nicely structured
+        # thus first check if that is available
+        try:
+            return self._download_country(
+                date_pub=date_pub_datetime,
+                clobber=clobber,
+            )
+        except ValueError:
+            try:
+                return self._download_region(
+                    date_pub=date_pub_datetime,
+                    clobber=clobber,
+                )
+            except ValueError:
+                raise RuntimeError(
+                    "No country or regional data found for"
+                    f" {date_pub_datetime.strftime('%Y-%m')}"
+                )
+
+    def process(self, *args, **kwargs):
+        """
+        Process FEWS NET data.
+
+        Method not implemented.
+        """
+        logger.info("`process()` method not yet implemented for FEWS NET.")
+
+    def load(  # type: ignore
```

Comment:
Hm, what do you think about putting a warning message in `load` saying that in future versions it will return the processed data? Will the two be very different or do you think it will be relatively backward compatible?

Reply:
I think it would change slightly significantly (aha). As now for the older data there is one shape polygon per IPC phase. And I would envision to have one polygon per livelihood zone (which is also what the newer raw data looks like). However, when looking at it nationally it thus should still contain the same information. 

But I like the idea of a warning message so will do that